be/src/io/cache/file_cache.cpp

```diff
@@ -0,0 +1,76 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "io/cache/file_cache.h"
+
+#include "common/config.h"
+#include "io/fs/local_file_system.h"
+
+namespace doris {
+namespace io {
+
+Status FileCache::download_cache_to_local(const Path& cache_file, const Path& cache_done_file,
+                                          io::FileReaderSPtr remote_file_reader, size_t req_size,
+                                          size_t offset) {
+    LOG(INFO) << "Download cache file from remote file: " << remote_file_reader->path().native()
+              << " -> " << cache_file.native();
+    io::FileWriterPtr file_writer;
```

Comment:
It is better to close file_writer and remote_file_reader when return an error.

Reply:
done